pinot-query-planner/src/main/java/org/apache/pinot/query/catalog/PinotCatalog.java

```diff
@@ -63,10 +64,12 @@ public PinotCatalog(TableCache tableCache) {
   public Table getTable(String name) {
     String tableName = TableNameBuilder.extractRawTableName(name);
     org.apache.pinot.spi.data.Schema schema = _tableCache.getSchema(tableName);
+    // TODO: This is not correct!
+    TableConfig tableConfig = _tableCache.getTableConfig(tableName + "_OFFLINE");
```

Comment:
The problem here is that TableCache is storing TableConfigs indexed by `tableNameWithType`, but we don't have the type here because Calcite calls this function and it doesn't know about offline and realtime tables.

Here I'm assuming the table will always have an offline part (which AFAIK is not something we can assume) and that in case it has both REALTIME and OFFLINE parts, both are going to be associated with the same table config (which I don't know if it is something we can assume).

The issue we want to solve here only makes sense in OFFLINE tables, but I think we may need to have the TableConfig here in future, so we should either find a solution where the table config is not necessary or find a solution where the table config will always be obtainable.

cc: @walterddr @Jackie-Jiang @xiangfu0